### PR TITLE
Throw exception if the database path directory is wrong

### PR DESCRIPTION
**Pull request by anuchak**
#1033 added a try catch block to handle incorrect database path directory

### CODE REVIEW

**Review by andyfengHKU — CHANGES_REQUESTED** [review]
Add tests on c++ and Python API

**Comment by anuchak** [comment]
added python & cpp api tests

File: tools/python_api/test/test_exception.py [hunk 1]
```diff
@@ -1,4 +1,7 @@
 import pytest
+import sys
+sys.path.append('../build/')
```
Comment (andyfengHKU) [inline tools/python_api/test/test_exception.py:3]: double check if this is needed.

**Review by andyfengHKU — APPROVED** [review]

File: tools/python_api/test/test_exception.py (hunk 1 above)
Comment (anuchak) [reply tools/python_api/test/test_exception.py:3]: yes seems to be required, the tests failed after I removed it: https://github.com/kuzudb/kuzu/actions/runs/3838919604/jobs/6535990703

**Pull request merged**